session_generation.go

```diff
@@ -100,3 +112,28 @@ func getVaultSession(store vaulted.Store, options *SessionOptions) (*vaulted.Ses
 	// Assume the session's role
 	return session.AssumeSessionRole()
 }
+
+func updateVaultFromEnvAndOptions(vault *vaulted.Vault, options *SessionOptions) {
+	// Calculate the region (lowest precedence to highest)
+	region := os.Getenv("AWS_DEFAULT_REGION")
```

Comment:
I would have put these in the opposite order but I'll allow it.

Reply:
I was trying to follow prior art :smile: :

* [Ruby SDK](https://github.com/aws/aws-sdk-ruby/blob/da245ccb2cf9260cee0c6822677d6caf738f7cc3/gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb#L50)
* [Go SDK](https://github.com/aws/aws-sdk-go/blob/60a7f0014f86142c03773706dd0cac474570e810/aws/session/env_config.go#L159)